rs/ic_os/network/src/info.rs

```diff
@@ -13,6 +14,27 @@ pub struct NetworkInfo {
     pub ipv6_address: Option<Ipv6Addr>,
     pub ipv6_subnet: u8,
     pub ipv6_gateway: Ipv6Addr,
+    pub ipv4_gateway: Option<Ipv4Addr>,
+    pub ipv4_prefix_length: Option<u8>,
+    pub domain: Option<String>,
+    pub mgmt_mac: Option<String>,
+}
+
+// TODO: NODE-1466: Remove in configuration revamp (HostOS and GuestOS integration)
+impl NetworkInfo {
+    pub fn to_network_settings(&self) -> NetworkSettings {
+        NetworkSettings {
+            ipv6_prefix: self.ipv6_prefix.clone(),
+            ipv6_address: self.ipv6_address,
+            ipv6_subnet: self.ipv6_subnet,
+            ipv6_gateway: self.ipv6_gateway,
+            ipv4_address: None,
+            ipv4_gateway: self.ipv4_gateway,
+            ipv4_prefix_length: self.ipv4_prefix_length,
+            domain: self.domain.clone(),
+            mgmt_mac: self.mgmt_mac.clone(),
+        }
+    }
```

Comment:
NetworkInfo is the old, `network/` internal networking struct. I've updated it to be as close to NetworkSettings as possible so that when we update HostOS to use the new config struct, we can replace NetworkInfo with NetworkSettings.

The only reason I didn't remove NetworkInfo is because I didn't want to mess with the way HostOS parses config.ini, which is tied to NetworkInfo. 

This way, we can update all the networking code to use NetworkSettings instead of NetworkInfo, but HostOS can still parse config.ini into a NetworkInfo object, which it then converts to a NetworkSettings object before calling the rest of the networking code.

Reply:
Update: I decided to go ahead and replace NetworkInfo with NetworkSettings entirely